apps/meteor/app/livechat/server/lib/RoutingManager.ts

```diff
@@ -101,17 +153,17 @@ export const RoutingManager = {
 		}
 
 		await LivechatRooms.changeAgentByRoomId(rid, agent);
-		await Rooms.incUsersCountById(rid);
+		await Rooms.incUsersCountById(rid, 1);
 
 		const user = await Users.findOneById(agent.agentId);
 		const room = await LivechatRooms.findOneById(rid);
 
-		await Message.saveSystemMessage('command', rid, 'connected', user);
+		user && (await Message.saveSystemMessage('command', rid, 'connected', user));
```

Comment:
this is the pattern we use, right?
```suggestion
		if (user) {
			await Message.saveSystemMessage('command', rid, 'connected', user));
		}
```